olp-cpp-sdk-dataservice-read/src/TaskSink.h

```diff
@@ -0,0 +1,96 @@
+/*
+ * Copyright (C) 2020 HERE Europe B.V.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ * License-Filename: LICENSE
+ */
+
+#pragma once
+
+#include <olp/core/client/CancellationToken.h>
+#include <olp/core/client/PendingRequests.h>
+#include <olp/core/client/TaskContext.h>
+#include <olp/core/logging/Log.h>
+#include <olp/core/thread/TaskScheduler.h>
+
+#include "repositories/ExecuteOrSchedule.inl"
+
+namespace olp {
+namespace dataservice {
+namespace read {
+
+class TaskSink {
+ public:
+  virtual bool Put(client::TaskContext task,
+                   thread::Priority priority = thread::NORMAL) = 0;
+};
+
+class DefaultTaskSink : public TaskSink {
+ public:
+  explicit DefaultTaskSink(
+      std::shared_ptr<thread::TaskScheduler> task_scheduler)
+      : task_scheduler_(std::move(task_scheduler)),
+        pending_requests_(std::make_shared<client::PendingRequests>()),
+        closed_(false) {}
+
+  DefaultTaskSink(const DefaultTaskSink&) = delete;
+  DefaultTaskSink& operator=(const DefaultTaskSink&) = delete;
+
+  ~DefaultTaskSink() {
+    closed_.store(true);
+    pending_requests_->CancelAllAndWait();
+  }
+
+  void CancelTasks() { pending_requests_->CancelAll(); }
+
+  bool Put(client::TaskContext task, thread::Priority priority) override {
```

Comment:
```suggestion
  bool Put(client::TaskContext task, uint32_t priority) override {
```
Why should we be able to only add three priorities?

Reply:
Reverted